src/main/scala/PlutoUpdaterActor.scala

```diff
@@ -0,0 +1,122 @@
+  private val proto="http"
+
+  private implicit val logger:DiagnosticLoggingAdapter = Logging.getLogger(this)
+
+  override def receive: Receive = {
+    case DoUpdate(atom)=>
+      logger.info(s"Received update request from ${sender()}")
+      val origSender = sender() //need to make a copy of this so when it's called from another thread in map() below it is passed correctly.
+      doUpdate(atom).map({
+      case Success(futureString)=>
+        futureString.onComplete({
+          case Success(serverResponse)=>
+            logger.info(s"Successfully updated vidispine: $serverResponse")
+            origSender ! SuccessfulSend
+          case Failure(error)=>
+            logger.warning(s"Updated vidispine but not able to read any response back")
+            origSender ! NoResponseBody
+        })
+      case Failure(error)=>
+        logger.error(error,s"Unable to update Vidispine: ${error.getMessage}")
+        origSender ! error
+    })
+    case _=>logger.error(s"Received an unknown message")
+  }
+
+
+  private def authString:String = Base64.getEncoder.encodeToString(s"$plutoUser:$plutoPass".getBytes)
+
+  private def send(itemId:String, xmlString:String):Future[Response[Source[ByteString, Any]]] = {
+    val bs = ByteString(xmlString,"UTF-8")
+
+    logger.info("Got xml body")
+    val source:Source[ByteString, Any] = Source.single(bs)
+    val hdr = Map(
+      "Accept"->"application/xml",
+      "Authorization"->s"Basic $authString",
+      "Content-Type"->"application/xml"
+    )
+
+    logger.info(hdr.toString())
+    val uri=uri"$proto://$plutoHost:$plutoPort/API/item/$itemId/metadata"
+    logger.info(s"Got headers, initiating send to $uri")
+    sttp
+      .put(uri)
+      .streamBody(source)
+      .headers(hdr)
+      .response(asStream[Source[ByteString, Any]])
+      .send()
+  }
+
+  private def consumeSource(source:Source[ByteString,Any]):Future[String] = {
+    logger.info("Consuming returned body")
+    val sink = Sink.reduce((acc:ByteString, unit:ByteString)=>acc.concat(unit))
+    val runnable = source.toMat(sink)(Keep.right)
+    runnable.run().map(_.utf8String)
+  }
+
+  def request(itemId:String, xmlString:String):Future[Try[Future[String]]] = send(itemId, xmlString).map({response=>
+    response.body match {
+    case Right(source)=>
+      logger.info("Send succeeded")
+      Success(consumeSource(source))
+    case Left(errorString)=>
+      VSError.fromXml(errorString) match {
+        case Left(unparseableError)=>
+          val errMsg = s"Send failed: ${response.code} - $errorString"
+          logger.warning(errMsg)
+          Failure(ErrorSend(errMsg))
+        case Right(vsError)=>
+          logger.warning(vsError.toString)
+          Failure(ErrorSend(vsError.toString))
+      }
+  }})
+
+  def getMasterId(atom:Atom):Option[String] = atom.data.asInstanceOf[AtomData.Media].media.metadata.flatMap(_.pluto.flatMap(_.masterId))
+
+  def doUpdate(atom:Atom):Future[Try[Future[String]]] = {
```

Comment:
I think we can make the type of the `atom` param `AtomData.Media` as we know its that type [here](https://github.com/guardian/multimedia-launchdetector-v3/pull/2/files#diff-0f7307c57e8188e6823e612cad21c1dcR87)

Reply:
I have done this where possible but am using root fields like ID and content change details so need the whole object